doc/man3/SSL_shutdown.pod

```diff
@@ -21,161 +21,267 @@ SSL_shutdown, SSL_shutdown_ex - shut down a TLS/SSL or QUIC connection
 
 =head1 QUIC-SPECIFIC SHUTDOWN CONSIDERATIONS
 
-When using QUIC, SSL_shutdown() or SSL_shutdown_ex() causes any data written to
-a stream which has not yet been sent to the peer to be written before the
-shutdown process is considered complete. An exception to this is streams which
-terminated in a non-normal fashion, for example due to a stream reset; only
-streams which are non-terminated or which terminated in a normal fashion have
-their pending send buffers flushed in this manner. This behaviour can be skipped
-by setting the B<SSL_SHUTDOWN_FLAG_NO_STREAM_FLUSH> flag; in this case, data
-remaining in stream send buffers may not be transmitted to the peer. This flag
-may be used when a non-normal application condition has occurred and the
-delivery of data written to streams via L<SSL_write(3)> is no longer relevant.
+When used with a QUIC connection SSL object, SSL_shutdown() initiates a QUIC
+immediate close using QUIC B<CONNECTION_CLOSE> frames.
+
+SSL_shutdown() cannot be used on QUIC stream SSL objects. To conclude a stream
+normally, see L<SSL_stream_conclude(3)>; to perform a non-normal stream
+termination, see L<SSL_stream_reset(3)>.
+
+SSL_shutdown_ex() may be used instead of SSL_shutdown() by an application to
+provide additional information to the peer on the reason why a connection is
+being shut down. The information which can be provided is as follows:
+
+=over 4
+
+=item I<quic_error_code>
+
+An optional 62-bit application error code to be signalled to the peer. The value
+must be in the range [0, 2**62-1], else the call to SSL_shutdown_ex() fails. If
+not provided, an error code of 0 is used by default.
+
+=item I<quic_reason>
+
+An optional zero-terminated (UTF-8) reason string to be signalled to the peer.
+The application is responsible for providing a valid UTF-8 string and OpenSSL
+will not validate the string. If a reason is not provided, or SSL_shutdown() is
+used, a zero-length string is used as the reason. If provided, the reason string
+is copied and stored inside the QUIC connection SSL object and need not remain
+allocated after the call to SSL_shutdown_ex() returns. Reason strings are
+bounded by the path MTU and may be silently truncated if they are too long to
+fit in a QUIC packet.
+
+Reason strings are intended for human diagnostic purposes only, and should not
+be used for application signalling.
+
+=back
+
+The arguments to SSL_shutdown_ex() are used only on the first call to
+SSL_shutdown_ex() (or SSL_shutdown()) for a given QUIC connection SSL object.
+These arguments are ignored on subsequent calls.
+
+These functions do not affect an underlying network BIO or the resource it
+represents; for example, a UDP datagram provided to a QUIC connection as the
+network BIO will remain open.
+
+=head2 Application Data Drainage Behaviour
+
+When using QUIC, SSL_shutdown() or SSL_shutdown_ex() ordinarily waits until all
+data written to a stream by an application has been acknowledged by the peer. In
```

Comment:
This reads as SSL_shutdown() is blocking, which it's not since it can return 0 to say it's not complete yet.